InAppUtils/InAppUtils.m

```diff
@@ -117,12 +117,15 @@ - (void)paymentQueueRestoreCompletedTransactionsFinished:(SKPaymentQueue *)queue
         NSMutableArray *productsArrayForJS = [NSMutableArray array];
         for(SKPaymentTransaction *transaction in queue.transactions){
             if(transaction.transactionState == SKPaymentTransactionStateRestored) {
-              NSDictionary *purchase = @{
-                @"originalTransactionIdentifier": transaction.originalTransaction.transactionIdentifier,
-                @"transactionIdentifier": transaction.transactionIdentifier,
-                @"productIdentifier": transaction.payment.productIdentifier,
-                @"transactionReceipt": [[transaction transactionReceipt] base64EncodedStringWithOptions:0]
-              };
+                SKPaymentTransaction *originalTransaction = transaction.originalTransaction;
+
+                NSDictionary *purchase = @{
+                    @"originalTransactionDate": @(originalTransaction.transactionDate.timeIntervalSince1970 * 1000),
```

Comment:
I think a number is simpler, cheaper and immediately comparable if needed. The only reasonable string format would be ISO-8601, and that requires using (admittedly simple) custom `NSDateFormatter` since `ISO8601DateFormatter` was only introduced in iOS 10. Then on the other side in JS, the `Date` constructor only supports a simplified version of ISO-8601 (see [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date)), which we'd need to make sure to use. String dates just feel more fragile and not as simple as using the standard milliseconds since epoch timestamps.